torch/csrc/distributed/autograd/functions/dist_accumulate_grad.cpp

```diff
@@ -0,0 +1,49 @@
+#include <torch/csrc/distributed/autograd/functions/dist_accumulate_grad.h>
+
+#include <torch/csrc/autograd/functions/accumulate_grad.h>
+#include <torch/csrc/autograd/functions/utils.h>
+#include <torch/csrc/autograd/variable.h>
+
+namespace torch::distributed::autograd {
+
+DistAccumulateGrad::DistAccumulateGrad(
+    AccumulateGrad&& accumulateGrad,
+    std::shared_ptr<DistAutogradContext> autogradContext)
+    : Node(accumulateGrad.sequence_nr()),
+      variable_(std::move(accumulateGrad.variable)),
+      autogradContext_(std::move(autogradContext)) {
+  move_from(std::move(accumulateGrad));
```

Comment:
can we maintain a pointer to AccumulateGrad node in DistAccumulateGrad, and use those contents in AccumulateGrad through the pointer?

Reply:
Delegating to `AccumulateGrad` should be better, but `Node:: post_hooks()` needs to be virtual (it's not right now). Making it(together with other Node functions) virtual may cause performance issue.